src/makeInfinityAdder.js

```diff
@@ -6,7 +6,22 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  const infinityAdder = (value) => {
+    if (value === undefined) {
+      const reserveSum = sum;
+
+      sum = 0;
+
+      return reserveSum;
+    } else {
+      sum += value;
+
+      return infinityAdder;
+    }
```

Comment:
else is redundant here
```suggestion
    sum += value;

    return infinityAdder;
```